### ISSUE TITLE
WithHealthzEndpoint should answer with application/json Content-Type

### ISSUE THREAD

**Issue by GreyXor** [grpc-ecosystem/grpc-gateway#2611]
## 🐛 Bug Report

`WithHealthzEndpoint` didn't set Content-Type to anything but `text/plain; charset=utf-8`. Shouldn't it be application/json?

## To Reproduce

1. Enable WithHealthzEndpoint
2. GET to the /healthz endpoint

## Expected behavior

```
Content-Type: application/json
Grpc-Metadata-Content-Type: application/grpc
```

## Actual Behavior

```
Content-Type: text/plain; charset=utf-8
```

## Your Environment
Go 1.18 with grpc-gateway v2.10.0
Linux 5.17.0-arch1-1 #1 SMP PREEMPT Tue, 22 Mar 2022 23:36:09 +0000 x86_64 GNU/Linux

**Comment by johanbrandhorst** [comment]
Yeah this definitely seems reasonable. Would you like to contribute this fix?

**Comment by johanbrandhorst** [comment]
CC @brumhard @antonioiubatti93

**Comment by GreyXor** [comment]
> Yeah this definitely seems reasonable. Would you like to contribute this fix?

Sure, but i didn't know where to define the headers. I never looked at the code of grpc-gateway. I'll be looking at

**Comment by GreyXor** [comment]
Since we stay in the [WithHealthEndpointAt()](https://github.com/grpc-ecosystem/grpc-gateway/blob/d8e890717a850f4bcc23f192fb567176d7f92e1b/runtime/mux.go#L216) method and we use the [outgoing marshaler](https://github.com/grpc-ecosystem/grpc-gateway/blob/d8e890717a850f4bcc23f192fb567176d7f92e1b/runtime/mux.go#L222) directly in it, we can't take advantage of the Content-Type deduction (except in the [errorHandler](https://github.com/grpc-ecosystem/grpc-gateway/blob/d8e890717a850f4bcc23f192fb567176d7f92e1b/runtime/mux.go#L241)). In my opinion it is easier in this case to directly add the headers [before](https://github.com/grpc-ecosystem/grpc-gateway/blob/d8e890717a850f4bcc23f192fb567176d7f92e1b/runtime/mux.go#L231) writing the response writer. 

I invite you to review my PR with the proposed changes. I'm listening to your comments
